tensorflow/contrib/go/tensor.go

```diff
@@ -0,0 +1,769 @@
+		t.StringVal = append(t.StringVal, resultBytes)
+	}
+	t.Dtype = pb.DataType(TF_TensorType(t.tensor))
+
+	return t.StringVal, nil
+}
+
+// Float32 returns the Tensor content as float32 slice if the tensor
+// type matches, if not returns a ErrInvalidTensorType error.
+// The datatypes are:
+//  - DTFloat
+func (t *Tensor) Float32() ([]float32, error) {
+	if DTFloat != t.DataType() {
+		return nil, &ErrInvalidTensorType{
+			tensorType:   t.DataType(),
+			expectedType: DTFloat,
+		}
+	}
+
+	if t.FloatVal != nil {
+		return t.FloatVal, nil
+	}
+
+	data := t.Data()
+	numElems := len(data) / cBytesFloat32
+	t.FloatVal = make([]float32, numElems)
+	for i := 0; i < numElems; i++ {
+		t.FloatVal[i] = math.Float32frombits(binary.LittleEndian.Uint32(data[i*cBytesFloat32 : (i+1)*cBytesFloat32]))
+	}
+	t.Dtype = pb.DataType(TF_TensorType(t.tensor))
+
+	return t.FloatVal, nil
+}
+
+// Float64 returns the Tensor content as float64 slice if the tensor
+// type matches, if not returns a ErrInvalidTensorType error.
+// The datatypes are:
+//  - DTDouble
+func (t *Tensor) Float64() ([]float64, error) {
+	if DTDouble != t.DataType() {
+		return nil, &ErrInvalidTensorType{
+			tensorType:   t.DataType(),
+			expectedType: DTDouble,
+		}
+	}
+
+	if t.DoubleVal != nil {
+		return t.DoubleVal, nil
+	}
+
+	data := t.Data()
+	numElems := len(data) / cBytesFloat64
+	t.DoubleVal = make([]float64, numElems)
+	for i := 0; i < numElems; i++ {
+		t.DoubleVal[i] = math.Float64frombits(binary.LittleEndian.Uint64(data[i*cBytesFloat64 : (i+1)*cBytesFloat64]))
+	}
+	t.Dtype = pb.DataType(TF_TensorType(t.tensor))
+
+	return t.DoubleVal, nil
+}
+
+// Int32 returns the Tensor content as int32 slice if the tensor
+// type matches, if not returns a ErrInvalidTensorType error.
+// The datatypes are:
+//  - DTUint8
+//  - DTInt8
+//  - DTInt16
+//  - DTInt32
+func (t *Tensor) Int32() ([]int32, error) {
+	if t.IntVal != nil {
+		return t.IntVal, nil
+	}
+
+	data := t.Data()
+	switch t.DataType() {
+	case DTInt8, DTUint8:
+		t.IntVal = make([]int32, len(data))
+		for i, v := range data {
+			t.IntVal[i] = int32(v)
+		}
+	case DTInt16:
+		numElems := len(data) / cBytesUint16
+		t.IntVal = make([]int32, numElems)
+		for i := 0; i < numElems; i++ {
+			t.IntVal[i] = int32(binary.LittleEndian.Uint16(data[i*cBytesUint16 : (i+1)*cBytesUint16]))
+		}
+	case DTInt32:
+		numElems := len(data) / cBytesInt32
+		t.IntVal = make([]int32, numElems)
+		for i := 0; i < numElems; i++ {
+			t.IntVal[i] = int32(binary.LittleEndian.Uint32(data[i*cBytesInt32 : (i+1)*cBytesInt32]))
+		}
+	default:
+		return nil, &ErrInvalidTensorType{
+			tensorType:   t.DataType(),
+			expectedType: DTInt32,
+		}
+	}
+
+	t.Dtype = pb.DataType(TF_TensorType(t.tensor))
+
+	return t.IntVal, nil
+}
+
+// Int64 returns the Tensor content as int64 slice if the tensor
+// type matches, if not returns a ErrInvalidTensorType error.
+// The datatypes are:
+//  - DTInt64
+func (t *Tensor) Int64() ([]int64, error) {
```

Comment:
Hah - I see I duplicated this discussion.  Int64s for sure, since rakyll also favors that.
(the plural, not "Slice" -- just like Bytes indicates many bytes, i.e., a slice of bytes)